testrunner-running/src/main/scala/edu/illinois/cs/testrunner/execution/TestListener.java

```diff
@@ -58,5 +78,55 @@ public void testFinished(Description description) throws Exception {
         } else {
             System.out.println("Test finished but did not start: " + fullTestName);
         }
+
+        String phase = Configuration.config().getProperty("statecapture.phase", "");
+        String order = Configuration.config().getProperty("statecapture.state", "");
+        String polluter = Configuration.config().getProperty("statecapture.polluter", "");
+        String fieldName = Configuration.config().getProperty("statecapture.fieldName", "");
+        System.out.println("PHASE: " + phase);
+
+        if (phase.equals("capture_after")) {
+            if (polluter.equals(fullTestName)) {
+                StateCapture sc = new StateCapture(fullTestName);
+                System.out.println("MainAgent.targetTestName: " + Configuration.config().getProperty("replay.dtname") +
+                        " fullTestName: " + fullTestName);
+                System.out.println("phase: " + phase);
+                System.out.println("order: " + order);
+                System.out.println("polluter: " + polluter);
+                System.out.println("test listener at after!!!!!!!!! Capturing the states!!!!!!!!!!!!!");
+                sc.capture();
+            }
+            else if (Configuration.config().getProperty("replay.dtname").equals(fullTestName)) {
+                System.out.println("MainAgent.targetTestName: " + Configuration.config().getProperty("replay.dtname") +
+                        " fullTestName: " + fullTestName);
+                System.out.println("phase: " + phase);
+                System.out.println("state: " + order);
+
+                StateCapture sc = new StateCapture(fullTestName);//CaptureFactory.StateCapture(fullTestName);
+                System.out.println("test listener at after !!!!!!!!! Capturing the states!!!!!!!!!!!!!");
+                sc.capture();
+            }
+        }
+        else if (phase.equals("load")) {
+            if (polluter.equals(fullTestName)) {
+                // reflect one field each time
+                if (fieldName.split(",").length == 1) {
```

Comment:
Perhaps should assign the `fieldName` here when it's going to be used as opposed to earlier.